datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -578,7 +579,13 @@ impl GroupedHashAggregateStream {
 /// [`GroupsAccumulatorAdapter`] if not.
 pub(crate) fn create_group_accumulator(
     agg_expr: &AggregateFunctionExpr,
+    group_by: &PhysicalGroupBy,
 ) -> Result<Box<dyn GroupsAccumulator>> {
+    // GROUPING is a special fxn that exposes info about group organization
+    if let Some(grouping) = agg_expr.fun().inner().as_any().downcast_ref::<Grouping>() {
+        let args = agg_expr.all_expressions().args;
+        return grouping.create_grouping_accumulator(&args, &group_by.expr);
+    }
```

Comment:
If we need special handling like this it seems to me that we should consider just making `Grouping` a build in. 

Or we should probably make it more generic so it can be used to implement some other function.  But since the input is is just the bitmaks and the output is the same. I wonder if there are any conceivable functions that could not just be implemented as a transformation on a builtin grouping function.

Reply:
It's kind of in a weird place, it's sort of not a real aggregation function but instead a way to leak metadata. That might be a reason to make it a built-in.

Do you have ideas about how and when to go about doing that?

There's another function called GROUP_ID (not to be confused with GROUPING_ID) which disambiguates duplicate rows, it might be relevant.

Reply:
> It's kind of in a weird place, it's sort of not a real aggregation function but instead a way to leak metadata. That might be a reason to make it a built-in.

> Do you have ideas about how and when to go about doing that?

One way might be that we expose the `grouping_id` column used in https://github.com/apache/datafusion/pull/12571 and implement the function as transformation on that. This should be possible as that column should "leak" the needed metadata. This is what was proposed in https://github.com/apache/datafusion/pull/5749

Reply:
I pushed an initial implemenation of this here: https://github.com/apache/datafusion/pull/12704

I think someone with more experince with this project should decide what is the best way forward.